clang/lib/Driver/ToolChains/Repo.h

```diff
@@ -0,0 +1,84 @@
+//===--- Repo.h - Repo ToolChain Implementations ----------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_DRIVER_TOOLCHAINS_REPO_H
+#define LLVM_CLANG_LIB_DRIVER_TOOLCHAINS_REPO_H
+
+#include "Linux.h"
+#include "clang/Driver/Tool.h"
+#include "clang/Driver/ToolChain.h"
+
+namespace clang {
+namespace driver {
+namespace tools {
+namespace repo {
+
+/// A class to directly call repo linker when targeting on *-repo.
+class LLVM_LIBRARY_VISIBILITY Linker : public Tool {
+public:
+  Linker(const ToolChain &TC) : Tool("repo::Linker", "rld", TC) {}
+
+  bool hasIntegratedCPP() const override { return false; }
+  bool isLinkJob() const override { return true; }
+
+  void ConstructJob(Compilation &C, const JobAction &JA,
+                    const InputInfo &Output, const InputInfoList &Inputs,
+                    const llvm::opt::ArgList &TCArgs,
+                    const char *LinkingOutput) const override;
+};
+} // namespace repo
+} // namespace tools
+
+namespace toolchains {
+
+/// RepoMuslToolChain - A tool chain using the repo tools and repo-based
+/// musl-libc and llvm libc++ libraries to perform compilation and linking
+/// commands. Currently does not support repo-based glibc library but
+/// compilation is fine.
+/// Since the *-musl-* toolchain doesn't support in the upstream LLVM, we need
+/// to add this class when targetting on repo and using musl libc linrary. We
```

Comment:
> Since the *-musl-* toolchain doesn't support in the upstream LLVM

Should be ”…toolchain isn’t supported by upstream…”

> when targetting on repo and using musl libc linrary

Should be ”when targeting the repo and using the musl libc library”.

Reply:
Fixed it in the commit [7c20e553](https://github.com/SNSystems/llvm-project-prepo/pull/158/commits/7c20e55309a449948d30e3fbcc7ffe4ed3361198).